### DISCUSSION TITLE
Add or Update Button while using Action Rule

### DISCUSSION THREAD

**Discussion by abhaypithadiya** [Power-Components/livewire-powergrid#617] [Q&A] [answered]
Hey! I have a delete button in my table I want to change that button to restore button when I use the soft deletes filter. How can I do that?

**Comment by dansysanalyst** [comment, marked as answer]
You can have a Delete and a Restore button and hide them accordingly to your Model state.

```php

use PowerComponents\LivewirePowerGrid\Rules\Rule;
use PowerComponents\LivewirePowerGrid\Header;

final class DishTable extends PowerGridComponent
{

//...

    public function actionRules(): array
    {
        return [

           //Hides delete button when row is trashed
            Rule::button('delete')
              ->when(fn ($dish) => $dish->trashed() == true)
              ->hide(),

           //Hides restore button when row is not trashed
            Rule::button('restore')
              ->when(fn ($dish) => $dish->trashed() == false)
              ->hide(),

         //Change row color for trashed rows
            Rule::rows()
              ->when(fn ($dish) => $dish->trashed() == true)
              ->setAttribute('class', 'bg-red-100 hover:bg-red-200'),
        ];
    }

    public function setUp(): array
    {
        $this->showCheckBox();

        return [
            Header::make()
            ->showSoftDeletes()
            ->showSearchInput(),
            //...
        ];
    }

    public function datasource(): ?Builder
    {
        return Dish::withTrashed()->with(['category:id,name', 'kitchen']);
    }
//...
}
```

Resulting in:

<img width="1186" src="https://user-images.githubusercontent.com/79267265/189072286-a4686f44-1508-46f1-8a80-6c0c348368f2.png">

**Reply by abhaypithadiya** [reply]
Hey! Thankyou for the quick answer. I have one more question. Can I hide the toggleable fields as well? If the record is deleted I don't want to show the toggleable field or can I use toggleable(false)?

**Reply by dansysanalyst** [reply]
This is currently not possible.

When mentioning this question to @luanfreitasdev, he also suggested using the setAttribute to modify the button behavior.

```php
setAttribute('wire:click', restore)
setAttribute('wire:click', delete)
```

**Reply by abhaypithadiya** [reply]
Hey! Thankyou again! This helped me!

**Reply by luanfreitasdev** [reply]
Hi @abhaypithadiya , the hide/show toggleable feature is now available in version 3.3.0

https://livewire-powergrid.com/#/table/action-rules?id=hidetoggleable

**Reply by abhaypithadiya** [reply]
Hey! I just saw it. Thankyou so much!
